### PR TITLE
AEA-1257 Multiple updates to confirmation aea/generic buyer due to AW4

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR adds two features in generic buyer/aw3:
- transaction behaviour is monitored, only if no transactions are left to process will the skill start new searches
- only the first 1000 characters of data are printed in generic buyer, avoiding spamming of logs.

## Fixes

- #2177

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: packages/fetchai/skills/confirmation_aw3/behaviours.py [hunk 1]
```diff
@@ -19,27 +19,11 @@
...
 from packages.fetchai.skills.generic_buyer.behaviours import (
     GenericSearchBehaviour,
     GenericTransactionBehaviour,
 )
 
 
 TransactionBehaviour = GenericTransactionBehaviour
-
-
-class SearchBehaviour(GenericSearchBehaviour):
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/confirmation_aw3/behaviours.py:34]: now done on generic buyer level

File: packages/fetchai/skills/confirmation_aw3/strategy.py [hunk 2]
```diff
@@ -109,9 +109,7 @@ def successful_trade_with_counterparty(
         """
         registration_db = cast(RegistrationDB, self.context.registration_db)
         registration_db.set_trade(counterparty, datetime.datetime.now(), data)
-        self.context.logger.info(
-            f"Successful trade with={counterparty}. Data acquired={data}!"
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/confirmation_aw3/strategy.py:113]: data already printed in handler of generic buyer

File: packages/fetchai/skills/generic_buyer/behaviours.py [hunk 3]
```diff
@@ -78,17 +78,28 @@ def act(self) -> None:
...
+        transaction_behaviour = cast(
+            GenericTransactionBehaviour, self.context.behaviours.transaction
+        )
+        remaining_transactions_count = len(transaction_behaviour.waiting)
+        if remaining_transactions_count > 0:
+            self.context.logger.info(
+                f"Transaction behaviour has {remaining_transactions_count} transactions remaining. Skipping search!"
             )
-            self.context.outbox.put_message(message=oef_search_msg)
+            return
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_buyer/behaviours.py:91]: new

File: packages/fetchai/skills/generic_buyer/behaviours.py [hunk 4]
```diff
@@ -78,17 +78,28 @@ def act(self) -> None:
...
+        strategy.update_search_query_params()
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_buyer/behaviours.py:92]: moved from aw3

File: packages/fetchai/skills/generic_buyer/handlers.py [hunk 5]
```diff
@@ -248,9 +248,8 @@ def _handle_inform(
         )
         if len(fipa_msg.info.keys()) >= 1:
             data = fipa_msg.info
-            self.context.logger.info(
-                "received the following data={}".format(pprint.pformat(data))
-            )
+            data_string = pprint.pformat(data)[:1000]
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_buyer/handlers.py:251]: avoid logs spamming

File: packages/fetchai/skills/generic_buyer/behaviours.py (hunk 3 above)
Comment (5A11) [reply packages/fetchai/skills/generic_buyer/behaviours.py:91]: Why is this?

Comment (DavidMinarsch) [reply packages/fetchai/skills/generic_buyer/behaviours.py:91]: Otherwise transactions pile up

**Review by 5A11 — APPROVED** [review]
LGTM

**Pull request merged**